mono/utils/mono-log-common.c

```diff
@@ -0,0 +1,143 @@
+/*
+ * mono-log-common.c: Platform-independent interface to the logger
+ *
+ * This module contains the POSIX syslog logger interface
+ *
+ * Author:
+ *    Neale Ferguson <neale@sinenomine.net>
+ *
+ */
+#include <config.h>
+
+#ifdef HAVE_UNISTD_H
+#include <unistd.h>
+#endif
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <ctype.h>
+#include <string.h>
+#include <glib.h>
+#include <errno.h>
+#include <time.h>
+#ifndef HOST_WIN32
+#include <sys/time.h>
+#endif
+#include "mono-logger.h"
+
+static FILE *logFile = NULL;
+static void *logUserData = NULL;
+
+/**
+ * mapSyslogLevel:
+ * 	
+ * 	@level - GLogLevelFlags value
+ * 	@returns The equivalent character identifier
+ */
+static inline char 
+mapLogFileLevel(GLogLevelFlags level) 
+{
+	if (level & G_LOG_LEVEL_ERROR)
+		return ('E');
+	if (level & G_LOG_LEVEL_CRITICAL)
+		return ('C');
+	if (level & G_LOG_LEVEL_WARNING)
+		return ('W');
+	if (level & G_LOG_LEVEL_MESSAGE)
+		return ('N');
+	if (level & G_LOG_LEVEL_INFO)
+		return ('I');
+	if (level & G_LOG_LEVEL_DEBUG)
+		return ('D');
+	return ('I');
+}
+
+/**
+ * mono_log_open_logfile
+ * 	
+ *	Open the logfile. If the path is not specified default to stdout. If the
+ *	open fails issue a warning and use stdout as the log file destination.
+ *
+ * 	@path - Path for log file
+ * 	@userData - Not used
+ */
+void
+mono_log_open_logfile(const char *path, void *userData)
+{
+	if (path == NULL) {
+		logFile = stdout;
+	} else {
+		logFile = fopen(path, "w");
+		if (logFile == NULL) {
+			g_warning("opening of log file %s failed with %s - defaulting to stdout", 
```

Comment:
It now defaults to stdout.

From: Johan Lorensson +ADw-notifications+AEA-github.com+ADw-mailto:notifications+AEA-github.com+AD4APg-
Reply-To: mono/mono +ADw-reply++0004b6fa40e51fb52a293a7d9a6e4cbca53d0f75c750d55292cf0000000113690ea392a170ce03ea30AQA-reply.github.com+ADw-mailto:reply++0004b6fa40e51fb52a293a7d9a6e4cbca53d0f75c750d55292cf0000000113690ea392a170ce03ea30AQA-reply.github.com+AD4APg-
Date: Friday, June 3, 2016 at 5:23 AM
To: mono/mono +ADw-mono+AEA-noreply.github.com+ADw-mailto:mono+AEA-noreply.github.com+AD4APg-
Cc: Neale Ferguson +ADw-neale+AEA-sinenomine.net+ADw-mailto:neale+AEA-sinenomine.net+AD4APg-, Mention +ADw-mention+AEA-noreply.github.com+ADw-mailto:mention+AEA-noreply.github.com+AD4APg-
Subject: Re: +AFs-mono/mono+AF0- Enhance log tracing (+ACM-2734)


In mono/utils/mono-log-common.c+ADw-https://github.com/mono/mono/pull/2734+ACM-discussion+AF8-r65679452+AD4-:

+AD4-  +ACo-
+AD4-  +ACo-    Open the logfile. If the path is not specified default to stdout. If the
+AD4-  +ACo-    open fails issue a warning and use stdout as the log file destination.
+AD4-  +ACo-
+AD4-  +ACo-    +AEA-path - Path for log file
+AD4-  +ACo-    +AEA-userData - Not used
+AD4-  +ACo-/
+AD4- ++vo-
+AD4- ++moAXw-log+AF8-open+AF8-logfile(const char +ACo-path, void +ACo-userData)
+AD4- +AHs-
+AD4-       if (path +AD0APQ- NULL) +AHs-
+AD4-               logFile +AD0- stdout+ADs-
+AD4-       +AH0- else +AHs-
+AD4-               logFile +AD0- fopen(path, +ACI-w+ACI-)+ADs-
+AD4-               if (logFile +AD0APQ- NULL) +AHs-
+AD4-                       g+AF8-warning(+ACI-opening of log file +ACU-s failed with +ACU-s - defaulting to stdout+ACI-,


Does it default to stdout elsewhere in case of error or is that part missing or should not be done at all (incorrect logging)?

+IBQ-
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub+ADw-https://github.com/mono/mono/pull/2734/files/583d1a2580a5fc61d0002920c17c23fa71375c3d+ACM-r65679452+AD4-, or mute the thread+ADw-https://github.com/notifications/unsubscribe/AAS2-t1npNgztELKVUYI62+AF8AXw-OA9hzwzeks5qH+AF8-KjgaJpZM4HsGtY+AD4-.